### PR TITLE
feat: Update TagList Loading State with a shimmer loader

### PR DESCRIPTION
**Pull request by Golodhros**
### Summary of Changes
Adds a TagList shimmer loader component
Adds it to our TagList component

### Tests
Updated tests

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes, including screenshots of any UI changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/common/ShimmeringTagListLoader/styles.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
+@import 'variables';
+
+$shimmer-loader-tag-height: 36px;
+$shimmer-loader-tag-border-radius: 4px;
```
Comment (ttannis) [inline amundsen_application/static/js/components/common/ShimmeringTagListLoader/styles.scss:4]: nit: I assume these two values were chosen to match `TagInfo`, so we perhaps we could add some shared tag-info variables in `_variables-default` and use them both here and in `TagInfo` for easier update if we ever change the size.

File: amundsen_application/static/js/components/common/ShimmeringTagListLoader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}: ShimmeringTagItemProps) => {
+  return (
+    <span
+      className={`shimmer-tag-loader-item shimmer-tag-loader-item--${index} is-shimmer-animated`}
+    />
+  );
+};
+
+export interface ShimmeringTagListLoaderProps {
+  numItems?: number; // Max items would be 15
```
Comment (ttannis) [inline amundsen_application/static/js/components/common/ShimmeringTagListLoader/index.tsx:23]: Is max items concept from design? If so we could also enforce that in the code with `Math.max()`.

File: amundsen_application/static/js/components/common/ShimmeringTagListLoader/styles.scss (hunk 1 above)
Comment (Golodhros) [reply amundsen_application/static/js/components/common/ShimmeringTagListLoader/styles.scss:4]: Yeah, makes sense!
I will check those up, thanks!

File: amundsen_application/static/js/components/common/ShimmeringTagListLoader/index.tsx (hunk 2 above)
Comment (Golodhros) [reply amundsen_application/static/js/components/common/ShimmeringTagListLoader/index.tsx:23]: I actually tried to enforce it, but didn't know the best way to go about it.
Throwing an error didn't work as expected, any ideas of how to approach it?

File: amundsen_application/static/js/components/common/ShimmeringTagListLoader/styles.scss (hunk 1 above)
Comment (Golodhros) [reply amundsen_application/static/js/components/common/ShimmeringTagListLoader/styles.scss]: I was only able to reuse the border radius value.

File: amundsen_application/static/js/components/common/ShimmeringTagListLoader/index.tsx (hunk 2 above)
Comment (ttannis) [reply amundsen_application/static/js/components/common/ShimmeringTagListLoader/index.tsx:23]: By "enforce" I don't think we need to go as far as making it such that the app throws an error but
1. What happens if we try `numItems?: 1 | 2 | 3 | ... | 15`?
2. If that doesn't do anything at best my original idea was to enforce a maximum of 15 in the render logic.

Comment (Golodhros) [reply amundsen_application/static/js/components/common/ShimmeringTagListLoader/index.tsx:23]: Oh, that makes sense!
Thanks!

**Pull request merged**